lib/src/modules/gates.dart

```diff
@@ -792,7 +799,7 @@ class ReplicationOp extends Module
 
   /// Executes the functional behavior of this gate.
   void _execute() {
-    replicated.put(_input.value, fill: true);
+    replicated.put(List.filled(_multiplier, _input.value).swizzle());
```

Comment:
What if we add a `replicate` function to `LogicValue` so that we can just call it here, plus get the benefit of doing replication on `LogicValue`s too?